doc/vxlan/Overlay ECMP with BFD.md

```diff
@@ -0,0 +1,329 @@
+
+
+# Definitions/Abbreviation
+###### Table 1: Abbreviations
+|                          |                                |
+|--------------------------|--------------------------------|
+| BFD                      | Bidirectional Forwarding Detection       |
+| VNI                      | Vxlan Network Identifier       |
+| VTEP                     | Vxlan Tunnel End Point         |
+| VNet                     | Virtual Network                |
+
+
+# 1 Requirements Overview
+
+## 1.1 Usecase
+
+Below diagram captures the use-case. In this, ToR is a Tier0 device and Leaf is a Tier1 device. Vxlan tunnel is established from Leaf (Tier1) to a VTEP endpoint. ToR (Tier0), Spine (Tier3) are transit devices. 
+
+
+![](https://github.com/Azure/SONiC/blob/master/images/vxlan_hld/OverlayEcmp_UseCase.png)
+
+### Packet flow
+
+- The packets destined to the Tunnel Enpoint shall be Vxlan encapsulated by the Leaf (Tier1).
+- Return packet from the Tunnel Endpoint (LBs) back to Leaf may or may not be Vxlan encapsualted.
+- Some flows e.g. BFD over Vxlan shall require decapsulating Vxlan packets at Leaf. 
+
+## 1.2 Functional requirements
+
+At a high level the following should be supported:
+
+- Configure ECMP with Tunnel Nexthops (IPv4 and IPv6)
+- Support IPv6 tunnel that can support both IPv4 and IPv6 traffic 
+- Tunnel Endpoint monitoring via BFD
+- Add/Withdraw Nexthop based on Tunnel or Endpoint health
+
+## 1.3 CLI requirements
+- User should be able to show the Vnet routes
+- This is an enhancement to existing show command
+
+## 1.4 Warm Restart requirements
+No special handling for Warm restart support.
+
+# 2 Modules Design
+
+The following are the schema changes. 
+
+## 2.1 Config DB
+
+Existing Vxlan and Vnet tables. 
+
+### 2.1.1 VXLAN Table
+```
+VXLAN_TUNNEL|{{tunnel_name}} 
+    "src_ip": {{ip_address}} 
+    "dst_ip": {{ip_address}} (OPTIONAL)
+```
+### 2.1.2 VNET/Interface Table
+```
+VNET|{{vnet_name}} 
+    "vxlan_tunnel": {{tunnel_name}}
+    "vni": {{vni}} 
+    "scope": {{"default"}} (OPTIONAL)
+    "peer_list": {{vnet_name_list}} (OPTIONAL)
+    "advertise_prefix": {{false}} (OPTIONAL)
+```
+
+## 2.2 APP DB
+
+### VNET
+
+The following are the changes for Vnet Route table
+
+Existing:
+
+``` 
+VNET_ROUTE_TUNNEL_TABLE:{{vnet_name}}:{{prefix}} 
+    "endpoint": {{ip_address}} 
+    "mac_address":{{mac_address}} (OPTIONAL) 
+    "vni": {{vni}}(OPTIONAL) 
+```
+
+Proposed:
+```
+VNET_ROUTE_TUNNEL_TABLE:{{vnet_name}}:{{prefix}}  
+    "endpoint": {{ip_address1},{ip_address2},...} 
+    "endpoint_monitor": {{ip_address1},{ip_address2},...} (OPTIONAL) 
+    "mac_address":{{mac_address1},{mac_address2},...} (OPTIONAL)  
+    "vni": {{vni1},{vni2},...} (OPTIONAL) 
+    "weight": {{w1},{w2},...} (OPTIONAL) 
```

Comment:
good point @rameshsanth. Since this was an enhancement over the existing implementation and the current route ecmp schema follows a similar approach, we decided to go with this list option. vni/weight/mac is one-to-one matched to endpoint. Its optional field; so user can fully ignore this field or leave empty "comma" separated list for any endpoint that doesn't have a corresponding vni.. 
For e.g endpoint 20.0.0.1 below:

```
endpoint:"10.0.0.1,20.0.0.1,30.0.0.1"
vni:"10,,30" 
```